src/cascadia/TerminalSettingsModel/JsonUtils.h

```diff
@@ -696,8 +707,41 @@ namespace Microsoft::Terminal::Settings::Model::JsonUtils
     template<typename T, typename Converter>
     void SetValueForKey(Json::Value& json, std::string_view key, const T& target, Converter&& conv)
     {
-        // demand guarantees that it will return a value or throw an exception
-        *json.demand(&*key.cbegin(), (&*key.cbegin()) + key.size()) = conv.ToJson(target);
+        std::optional<Json::Value> out;

```

Comment:
I'm having a hard time wrapping my head around this code - do we have a good test case that we know tests this string/optional behavior?

Reply:
We specifically need this for `StartingDirectory`. If you set it to null in the JSON, we store it as the empty string. For ToJson, we know `_startingDirectory` is an hstring. So that means it is optional, and it is a string. So we pass the hstring to the converter (as opposed to the `optional<hstring>`). The empty string gets treated as an empty string again, and we write it out.

This'll be a good one for me to add as a test along with `foreground`, `tabColor` and maybe `source`. That way, we have all of these weird ones accounted for.

Reply:
I still don't 100% understand why StartingDirectory couldn't have been a `Nullable<hstring>`

Reply:
If I use `GETSET_NULLABLE_SETTING`, we'll try to expose it as an `IReference<hstring>` and that's not allowed :(. I would have to rewrite the getter/setter to make it truly nullable, but this weird-ish workaround works in the meantime. I'm open to some ideas on how to break it down more eloquently though.